modules/javafx.controls/src/main/java/javafx/scene/control/ControlUtils.java

```diff
@@ -167,6 +168,7 @@ private void checkState() {
             sm.startAtomic();
             final List<Integer> removed = c.getRemoved().stream()
                     .map(TablePositionBase::getRow)
+                    .filter(removeRowFilter)
                     .distinct()
```

Comment:
Maybe `distinct` should be applied before `filter`. This can cut down the number of times the predicate is invoked (which iterates over all selected cells, so it may be a performance issue for large selections).

Reply:
Makes sense. Done